### ISSUE TITLE
bypass notification

### ISSUE THREAD

**Issue by ghost** [uBlockOrigin/uAssets#5890]
### URL(s) where the issue occurs

`http://sprysphere.com/BiZZ`

### Describe the issue

The link above require notification to be allowed in order to continue but i don't want any notification. Can it be bypassed?

### Screenshot(s)

![image](https://user-images.githubusercontent.com/24647715/60387888-4688d080-9aaa-11e9-8127-3e014c46f036.png)

### Versions

- Browser/version: 1.20.0
- uBlock Origin version: Opera 60.0.3255.170

### Settings

- Default

### Notes

None

**Comment by krystian3w** [comment]
no filters block push...

**Comment by krystian3w** [comment]
@mapx-  badfilter:

`/ntfc.php?$script`

url:
````
https://www1.ecleneue.com/pushredirect/?placementid=10729309&clickid=5743133391&t=1561836513&dest=https%3A%2F%2Fwww.dropbox.com%2Fs%2Flwph95lsyuv21kw%2FLDOE_v1.8_Mod_iOSGods_Div.zip%3Fdl%3D0
```

**Comment by krystian3w** [comment]
@kenviet links open after 4 redirect in `maranhesduve.club` site.

**Comment by okiehsch** [comment]
After getting redirected through `maranhesduve.club` I arrive at
![image](https://user-images.githubusercontent.com/16838044/60389025-8c4d9500-9aba-11e9-8d1a-20b42ca8e686.png)

without any notification on my end.

**Comment by krystian3w** [comment]
`https://p1.maranhesduve.club/?tag_id=759228&sub_id1=10729309&sub_id2=5946832176255722117&cookie_id=1a5d90e7-dcc7-4ee2-9bf1-0d76f9a4ce5e&lp=adfly_allow&convert=Your%20Video%20Is%20Ready%20To%20Stream&allb=redirect&ob=redirect&href=https%3A%2F%2Fwww.dropbox.com%2Fs%2Flwph95lsyuv21kw%2FLDOE_v1.8_Mod_iOSGods_Div.zip%3Fdl%3D0&hop=-1&info=&sub=p1`

`https://p2.maranhesduve.club/?tag_id=759228&sub_id1=10729309&sub_id2=5946832176255722117&cookie_id=1a5d90e7-dcc7-4ee2-9bf1-0d76f9a4ce5e&lp=adfly_allow&convert=Your%20Video%20Is%20Ready%20To%20Stream&allb=redirect&ob=redirect&href=https%3A%2F%2Fwww.dropbox.com%2Fs%2Flwph95lsyuv21kw%2FLDOE_v1.8_Mod_iOSGods_Div.zip%3Fdl%3D0&hop=0&info=&sub=p2`

**Comment by krystian3w** [comment]
![obraz](https://user-images.githubusercontent.com/35370833/60389062-5c52c180-9abb-11e9-8c5b-5000455e3871.png)

**Comment by okiehsch** [comment]
````
https://p1.maranhesduve.club/?tag_id=759228&sub_id1=10729309&sub_id2=5946832176255722117&cookie_id=1a5d90e7-dcc7-4ee2-9bf1-0d76f9a4ce5e&lp=adfly_allow&convert=Your%20Video%20Is%20Ready%20To%20Stream&allb=redirect&ob=redirect&href=https%3A%2F%2Fwww.dropbox.com%2Fs%2Flwph95lsyuv21kw%2FLDOE_v1.8_Mod_iOSGods_Div.zip%3Fdl%3D0&hop=-1&info=&sub=p1
````

I do have those and I also get the screenshot of the OP but I get automatically redirected to the dropbox site without me having to do anything

**Comment by krystian3w** [comment]
for me site wait for allow push and push also blocked on sister domain:


```
https://www1.ecleneue.com/pushredirect/?placementid=10729309&clickid=5743133391&t=1561839614&dest=https%3A%2F%2Fwww.dropbox.com%2Fs%2Flwph95lsyuv21kw%2FLDOE_v1.8_Mod_iOSGods_Div.zip%3Fdl%3D0
```

**Comment by okiehsch** [comment]
Well, not for me and I would not badfilter `/ntfc.php?$script` maybe allow it for the site only if necessary.
Let's wait and see if **mapx-** can reproduce.

**Comment by mapx-** [comment]
I cannot arrive at dropbox even whitelisting
```
@@||pushokey.com/ntfc.php$script,domain=ecleneue.com
@@||pushlaram.com/ntfc.php$script,domain=ecleneue.com
```

**Comment by okiehsch** [comment]
Do you get the same result if you go to
`sprysphere.com/DvHm`?
The shortener leads to a `mediafire` site on my end.

**Comment by mapx-** [comment]
yeah, same behaviour the destination page (waiting for allow) is
`https://www1.ecleneue.com/pushredirect/?placementid=3860662&clickid=5744570877&t=1561845822&dest=https%3A%2F%2Fgloyah.net%2Fredirecting%2FaHR0cDovL3d3dy5tZWRpYWZpcmUuY29tL2ZvbGRlci9oYWpyMWZqd2cwb2IxL0ZSUDIwMTg%3D%2F2df3c7c1bae1ddeeeeea3e09ff221ae2`

**Comment by okiehsch** [comment]
I never get to `ecleneue.com`.
It's `sprysphere` --> `gloyah` --> `maranhesduve` --> `mediafire` on my end.

If it does not work at all on your end adding an exception for the push notification seems pointless.

**Comment by mapx-** [comment]
using a hungarian ip:
- FF => `sprysphere --> gloyah --> maranhesduve` (not arriving on `mediafire`)
- chrome => `sprysphere --> gloyah --> ecleneue` (not arriving on `mediafire`)

**Comment by okiehsch** [comment]
Does it work if you whitelist `maranhesduve` or `ecleneue`?

**Comment by mapx-** [comment]
No effect => still waiting `allow`

**Comment by okiehsch** [comment]
Well, then it seems to be a site issue on your end, though I don't know why it works for me and for **krystian3w** it works after adding the exceptions.

**Comment by krystian3w** [comment]
`maranhesduve.club##+js(abort-on-property-read.js, Notification.requestPermission)`

maybe works to disable webpush if for other users allow notify is no needed.

**Comment by dumbusernameidk** [comment]
I can't reproduce anything either opening the link in the op goes through correctly with no issue.

**Comment by ghajini** [comment]
I can reproduce on firefox android....
Site needs to allow `@@||pushokey.com^$script`
Then it drops you to Dropbox at the ***expense of 3rd party trackers exposure***
@jspenguin2017 can you help here

**Comment by jspenguin2017** [comment]
We need a new script snippet.

**Comment by jspenguin2017** [comment]
Eh... The notification is requested from a service worker... 
The only way I can think of interfering with it is to periodically revoke the notification permission for those bad domains.

**Comment by krystian3w** [comment]
Reddirection?

    FF => sprysphere --> gloyah --> maranhesduve > p4.maranhesduve.club
    chrome => sprysphere --> gloyah --> ecleneue  > www4.ecleneue.com

website should ignore notification after 4 tests.

**Comment by jspenguin2017** [comment]
Hum... Yea, if you do not grant notification it will try ~5 times then give up.

**Comment by krystian3w** [comment]
`https://www4.ecleneue.com/pushredirect/?placementid=10729309&clickid=5743133391&t=1561927096&dest=https%3A%2F%2Fwww.dropbox.com%2Fs%2Flwph95lsyuv21kw%2FLDOE_v1.8_Mod_iOSGods_Div.zip%3Fdl%3D0`

for me www4 no need:

```
`@@/ntfc.php?$script,domain=ecleneue.com`

**Comment by okiehsch** [comment]
> Hum... Yea, if you do not grant notification it will try ~5 times then give up.

That is what I get at my end, `maranhesduve` automatically reloads 4 times and then I get redirected to `dropbox` or `mediafire` in the two linked examples.

**Comment by ghajini** [comment]
Stuck at `please press allow to continue`
No matter how much I wait

**Comment by krystian3w** [comment]
For my `ecleneue.com` is broken and manualy I must change www**X** to www4.

`maranhesduve.com` I must deny push 3 times or manualy change p**X** to p1 (numbering down to lower)

**Comment by krystian3w** [comment]
In Firefox also have both domains `maranhesduve.club` and `ecleneue.com`

**Comment by krystian3w** [comment]
Maybe fix mara if my IP is not guilty:

```
! 1.07.2019 maranhesduve.club
||pixel.maranhesduve.club^$xmlhttprequest
||maranhesduve.club^$script
```

**Comment by kenviet** [comment]
idk whatever it's fixed. yesterday, I had to allow it in order to arrive at dropbox
I removed notification permission after but still let me arrive at it lol

**Comment by krystian3w** [comment]
Cookies / IP as suggested by @ghajini

**Comment by krystian3w** [comment]
Or block all scripts and xhr helps on `mara`.

**Comment by ghajini** [comment]
still stuck and iam waiting for more than 10 mins for site to autoredirect to destination site
`https://user-images.githubusercontent.com/20338483/60672562-567e2700-9e93-11e9-8bf0-239da508ea64.png`

source page served
`https://pastebin.com/nkdF7HSC`

**Comment by krystian3w** [comment]
`ecleneue.com` maybe still broken by `/ntfc.php?$script`

try:

`@@/ntfc.php?$script,domain=ecleneue.com`

**Comment by jspenguin2017** [comment]
@ghajini Huh, the destination URL is hard coded in the DOM? We can probably make a bypass for it...

**Comment by jspenguin2017** [comment]
I noted it down, I don't know when I can have time to look into this though, I'm too busy these weeks.

**Comment by ghajini** [comment]
Another such site

`https://ww10.ouo.today/?cr=aHR0cDovL3VsdGltYXRlLnR1cmtkb3duLmNvbS9saW5rLnBocD9saW5rPWFIUjBjSE02THk5Nk1ta3VZMjl0TDI5a1IweHhSMHQ2&oID=82904`

`https://user-images.githubusercontent.com/20338483/61239037-7e8c4680-a72d-11e9-847c-aceea5002d66.jpg`

**Comment by krystian3w** [comment]
Block script cannot bait this `ww10.ouo.today`...

**Comment by krystian3w** [comment]
https://github.com/AdguardTeam/AdguardFilters/issues/37022 similar.

**Comment by ghajini** [comment]
Adguard browser extension works with their  annoyance filters enabled.....
Similar/other way can be done ?
@okiehsch, @gorhill, @jspenguin2017

**Comment by krystian3w** [comment]
uBO no support custom script as filter and uBO may no add any new scriplet.

Try baypass addon/user script (ViolentMonkey/TampeMonkey)?

**Comment by jspenguin2017** [comment]
Oh, the destination is in GET parameter? Wow, how did I miss that.

**Comment by JobcenterTycoon** [comment]
i get redirected to `http://xterca.net/-36715UDWO/BiZZ?rndad=521214842-1564420884` and i get anti Adblock
![antiatblock](https://user-images.githubusercontent.com/48647394/62068306-1ada4100-b236-11e9-821c-25e30b6149bb.PNG)

**Comment by krystian3w** [comment]
`*/ad_top_*$image,first-party,redirect=1x1.gif`
`*/popunder.*$image,domain=xterca.net,redirect=1x1.gif`
`@@||d1nmxiiewlx627.cloudfront.net^$script,domain=xterca.net`

But better maybe one scriplet?

**Comment by scottmerck7007** [comment]
dismiss denial

**Comment by scottmerck7007** [comment]
clear

**Comment by ghajini** [comment]
Closed without any fix?

**Comment by krystian3w** [comment]
`xterca.net` gone as timeout counter with Anti-Adblock (IMO) or fixed silent anywhere (IDK).

`sprysphere.com` and other notification domains no worth for uBO resources?

**Comment by mapx-** [comment]
- it's like https://github.com/uBlockOrigin/uAssets/issues/6098
still opened
- only @gorhill could tell us if a scriptlet could fix such issues
- someone could test if a userscript like this 1
https://greasyfork.org/en/scripts/382794-oxy-cloud-download-helper/code

could help in this case too ?

**Comment by krystian3w** [comment]
`http://larati.net/-36715GEIC/BiZZ?rndad=637050862-1566237639`

same as https://github.com/uBlockOrigin/uAssets/issues/5890#issuecomment-516083618

![obraz](https://user-images.githubusercontent.com/35370833/63288104-36c48600-c2bc-11e9-845c-5a9f15bec334.png)


---

could help:

```
  */ad_top_*$1p,image,domain=xterca.net|larati.net,redirect=1x1.gif
*/popunder.*$3p,image,domain=xterca.net|larati.net,redirect=1x1.gif

@@||d1nmxiiewlx627.cloudfront.net^$script,domain=xterca.net|larati.net
*/*.php?$1p,subdocument,domain=xterca.net|larati.net,redirect=noop.html
```

**Comment by okiehsch** [comment]
`http://larati.net/-36715GEIC/BiZZ?rndad=637050862-1566237639`

Have you updated your `uBO-filters` list?

I end up at 
`https://www.dropbox.com/s/lwph95lsyuv21kw/LDOE_v1.8_Mod_iOSGods_Div.zip?dl=0`
without me doing anything.

**Comment by krystian3w** [comment]
ok on mobile works and again comeback to broken:

`https://www1.ecleneue.com/pushredirect/?placementid=10729309&clickid=5743133391&t=1566276790&dest=https%3A%2F%2Fwww.dropbox.com%2Fs%2Flwph95lsyuv21kw%2FLDOE_v1.8_Mod_iOSGods_Div.zip%3Fdl%3D0`

And `maranhesduve.club`

**Comment by ghajini** [comment]
Anyways we are stuck(with default updated filters) at `please press allow to continue`

**Comment by krystian3w** [comment]
Nope, for my block all scripts on `maranhesduve.club` disable push verification.

`ecleneue.com` broken by https://github.com/uBlockOrigin/uAssets/issues/5890#issuecomment-508545692 even probably on `www4.ecleneue.com` (4 she didn't check push anymore, she thought that you refused three times and AdFly finally let him pass).

**Comment by krystian3w** [comment]
Nope:

* for my: block almost all scripts on `maranhesduve.club` disable push verification.

* `ecleneue.com` broken by https://github.com/uBlockOrigin/uAssets/issues/5890#issuecomment-508545692 even probably on `www1.ecleneue.com`, `www2.ecleneue.com`, `www3.ecleneue.com` (`www4.ecleneue.com` didn't check push anymore, sp thought that you reject three times and AdFly finally let pass).

**Comment by krystian3w** [comment]
Nope:

* for my: block almost all scripts on `maranhesduve.club` disable push verification.

* `ecleneue.com` broken by https://github.com/uBlockOrigin/uAssets/issues/5890#issuecomment-508545692 even probably on `www1.ecleneue.com`, `www2.ecleneue.com`, `www3.ecleneue.com` (`www4.ecleneue.com` didn't check push anymore, so assumes you reject three times and AdFly finally let pass).

**Comment by lyndmi** [comment]
The workaround of this issue is to use Epic Privacy Browser, because it only remember notification settings temporary. If you closes it, it will forget that you allowed notification of a site and will no longer appear in Allow list.

**Comment by krystian3w** [comment]
IMO better `whitelist tribe in AdGuard` - works only on `maranhesduve.club` / `ecleneue.com` (problematic sites needed individual scripts or deactivating CSP in Firefox)

 if we don't use scriplets in `Violent/TamperMoney` or non-default uBO resources load from advanced settings.

**Comment by ghajini** [comment]
Similar site

`https://sparbuttantowa.pro/LRODTI?tag_id=777823&sub_id1=195668&sub_id2=6977632712313340712&cookie_id=f7919b46-c2e2-4e8e-b916-7f065a57bacd&lp=loading&tb=redirect&allb=redirect&ob=redirect&href=https%3A%2F%2Freroplittrewheck.pro%2F%3Ftid%3D778547%26noocp%3D1%26subid%3D195668&hop=7`

**Comment by mapx-** [comment]
adding here the other issue (`oxy.cloud`)
https://github.com/uBlockOrigin/uAssets/issues/6098

**Comment by mapx-** [comment]
other issue:
https://github.com/uBlockOrigin/uAssets/issues/6491

**Comment by krystian3w** [comment]
New domain with same problem:

`https://www1.debrahinton.pro/pushredirect/?site=adfly&network=1&ppi=4264276&pci=2458551268&t=1573403816&dest=https%3A%2F%2Fcowner.net%2Fredirecting%2FaHR0cDovL3d3dy5tZWRpYWZpcmUuY29tL2Rvd25sb2FkL2xkM2JmbzIyN21qbW9ray9OZXZlcmVuZGluZ19Ub3dlci56aXA%3D%2F1eedd59c2840350d9bf7d87f0df4a43d`

**Comment by krystian3w** [comment]
New domains with same problem:

`cowner.net` ➡️  `www1.debrahinton.pro` ➡️  `www2.lucienmann.pro`  ➡️  `www3.sherwoodsutton.pro`  ➡️  `www4.ramirocampos.pro`  ➡️  `cowner.net`  ➡️  e.g. mediafire download file.
 

<ul> <li><blockquote>

```
https://www1.debrahinton.pro/pushredirect/?site=adfly&network=1&ppi=4264276&pci=2458551268&t=1573403816&dest=https%3A%2F%2Fcowner.net%2Fredirecting%2FaHR0cDovL3d3dy5tZWRpYWZpcmUuY29tL2Rvd25sb2FkL2xkM2JmbzIyN21qbW9ray9OZXZlcmVuZGluZ19Ub3dlci56aXA%3D%2F1eedd59c2840350d9bf7d87f0df4a43d
```

no help block all scripts and xhr


</blockquote><li><blockquote>

```
https://www2.lucienmann.pro/pushredirect/?network=1&site=adfly&ppi=4264276&pci=2458551268&t=1573403816&dest=https%3A%2F%2Fcowner.net%2Fredirecting%2FaHR0cDovL3d3dy5tZWRpYWZpcmUuY29tL2Rvd25sb2FkL2xkM2JmbzIyN21qbW9ray9OZXZlcmVuZGluZ19Ub3dlci56aXA%3D%2F1eedd59c2840350d9bf7d87f0df4a43d
```

no tested block all xhr and scripts.

</blockquote><li><blockquote>

```
https://www3.sherwoodsutton.pro/pushredirect/?network=1&site=adfly&ppi=4264276&pci=2458551268&t=1573403816&dest=https%3A%2F%2Fcowner.net%2Fredirecting%2FaHR0cDovL3d3dy5tZWRpYWZpcmUuY29tL2Rvd25sb2FkL2xkM2JmbzIyN21qbW9ray9OZXZlcmVuZGluZ19Ub3dlci56aXA%3D%2F1eedd59c2840350d9bf7d87f0df4a43d
```

no tested block all xhr and scripts.

</blockquote><li><blockquote>

```
https://www4.ramirocampos.pro/pushredirect/?network=1&site=adfly&ppi=4264276&pci=2458551268&t=1573403816&dest=https%3A%2F%2Fcowner.net%2Fredirecting%2FaHR0cDovL3d3dy5tZWRpYWZpcmUuY29tL2Rvd25sb2FkL2xkM2JmbzIyN21qbW9ray9OZXZlcmVuZGluZ19Ub3dlci56aXA%3D%2F1eedd59c2840350d9bf7d87f0df4a43d
```

no tested block all xhr and scripts.

</blockquote></ul>

**Comment by jspenguin2017** [comment]
It looks like redirecting to `dest` doesn't work, there's probably other validations on server side. I'll still keep the script snippet though, might be useful in other cases. It's privileged for now, as I'm not sure if it can be misused.

**Comment by prateekrastogi** [comment]
Despite multiple issues to fix Bing Ads, there has been no response whatsoever and agitation from repo maintainers:
So, referencing my issue here:
https://github.com/uBlockOrigin/uAssets/issues/6887#issuecomment-578509029

Let's name and shame ublock origin until their ego issues get resolved.

**Comment by dumbusernameidk** [comment]
Your issue has already been resolved so I don't see why you need to spam it constantly.

**Comment by uBlock-user** [comment]
@mapx- `http://sprysphere.com/BiZZ` redirects to `gatustox.net` which was fixed recently. No need to keep it open ?

**Comment by mapx-** [comment]
well, I don't know if all other sites were fixed .. however we don't have for now a specific fix. I'd say we can close it for now.

**Comment by uBlock-user** [comment]
> if all other sites were fixed

what other sites ?

**Comment by mapx-** [comment]
`oxy.cloud` ? (see above)

**Comment by uBlock-user** [comment]
Go to `https://oxy.cloud/d/pyH`, Click on the free download blue button, 

Get redirected from  `1.news-fbe.com` and till `10.news-fbe.com` and then finally reach -- `https://oxy.cloud/d/pyH/2/9d9c30a7bad75647eebca17ade51869f`

Add `oxy.cloud##+js(ra, disabled)` to My Filters and refresh the page, wait till the bar get filled to 100 % and then click Download File blue button, and file gets downloaded.

Not seeing any issue here either except the disabled button which can be fixed with the above filter.

**Comment by mapx-** [comment]
ok then .. fix it and we can close the thread

**Comment by uBlock-user** [comment]
@mapx- Actually enabling the button back doesn't help, file will not be downloaded until the bar gets filled (100 %) and once the bar is filled the button does get enabled, so should we still enable it by force ?

**Comment by mapx-** [comment]
I still get

![image](https://user-images.githubusercontent.com/10303732/75762869-ac191500-5d3b-11ea-819f-c698e16a8b2e.png)

which is the original issue, right ?

**Comment by uBlock-user** [comment]
i get that too and i wait for a minute or less and I get redirected.

**Comment by mapx-** [comment]
I'm stuck on that page, so I cannot test further

**Comment by uBlock-user** [comment]
Is `gstatic.com` blocked when you get stuck on `news-fbe.com` ? I get stuck if I block `gstatic.com`

**Comment by mapx-** [comment]
no, nothing blocked, using chrome + default settings

**Comment by okiehsch** [comment]
> which is the original issue, right ?

Correct, I don't get stuck but let's leave the issue open. It's not like we are drowning in open issues.

**Comment by uBlock-user** [comment]
Must be a Chrome issue then, work on Firefox.

**Comment by ghajini** [comment]
Actually there are so many domains(leeching /premium link generator sites)  doing this ***please press allow to continue***
we are as user of ublock origin stuck on this and had to rely on ambiguous extension universal bypass to do this
Any fix available?

**Comment by LGLTeam** [comment]
This site have it too
`https://shre.su/F0NV`

**Comment by krystian3w** [comment]
for my generate 5-6 redirect on "`gumk.pro`"

end open `https://oxy.st/d/mAvb` with your apk  cracked (or modded) game / app.

**Comment by uBlock-user** [comment]
> This site have it too
https://shre.su/F0NV

Specific to Chrome as above, can't reproduce on Firefox.

**Comment by ghajini** [comment]
Any custom scriptlet that I can use at `userresourcelocation`
I have working solution which work in userscript manager but userscript manager is unsupported/unavailable  as of now in firefox android extension platform ....

URL =
`https://www42.elbaestes.pro/pushredirect/?network=1&site=adfly&ppi=2220594&pci=6707862119&t=1608455433&dest=https%3A%2F%2Fonleech.me%2Fshort%3Fomg%3DvrUMDPLgiiO-BEck9ry8DVovYO_tLrxKyOJUZcP2P_GKiECapmtnoCaoLZsFENe38x_5UsHKh6y-gRv4JDo9kE7g6L5w8T6GCF4NbjZLhENMkEDVSCfgq2IcG0DEbA18`

Got from `https://exe.app/U8JdoJM`

Issue= 
can't proceed further to destination site, stuck at site saying **Please press 'Allow' to continue**

![Screenshot_2020-12-20-14-45-03-616_org mozilla fenix](https://user-images.githubusercontent.com/20338483/102709714-c88a3c00-42d2-11eb-96ef-4c8ab7c00b8b.png)

Platform=
**firefox android stable *

Note=
Please consider fixing this in ublock origin itself @gwarser @gorhill

**Comment by krystian3w** [comment]
`grant-fake-notification` no help for these ad-fly short url engine = bugged by useage serviceWorker.

**Comment by IThinkIAmCool** [comment]
I added `darenjarvis.pro##+js(adfly-defuser)` to 'My filters' and then accessed the first link from the issue again.

![image](https://user-images.githubusercontent.com/84580704/120855864-b00f9480-c587-11eb-9f79-24a3ba0c2972.png)

**Comment by IThinkIAmCool** [comment]
![image](https://user-images.githubusercontent.com/84580704/120855986-d6cdcb00-c587-11eb-85a5-b61b0f3fb80c.png)

**Comment by krystian3w** [comment]
Maybe resource need update...?

**Comment by the-poor-gamer-124** [comment]
how about you instead stop using free resources? no more ads that way

**Comment by krystian3w** [comment]
May need add can't Fix or won't fix.

Also lock conversation to collaborators only is good idea.

**Comment by the-poor-gamer-124** [comment]
freeloaders not liking the truth
